File: Lab2/Lab_2_2.java
public class Main
{   
    public static String integerToRoman(int n) {
    String[] a = new String[]{"I","IV","V","IX","X","XL","L","XC","C","CD","D","CM","M"};
    int[] b = new int[]{1,4,5,9,10,40,50,90,100,400,500,900,1000};
    StringBuilder s = new StringBuilder();
    for (int i = a.length - 1; i >= 0; i--) {
        while (n >= b[i]) {
            s.append(a[i]);
            n -= b[i];
        }
    }
    return s.toString();
}
	public static void main(String[] args) {
		System.out.println(integerToRoman(193));
		
	}
}

File: Lab2/Lab_2_2.cs
using System;
using System.Text;
					
public class Program
{
	public static string NumberToRoman(int number)
	{
		if (number < 0 or number > 3999)
    		throw new ArgumentException("enter value 0 - 3999");
		if (number == 0) return "N";
		
		int[] values = new int[] { 1000, 900, 500, 400, 100,90, 50, 40, 10, 9, 5, 4, 1 };
		string[] numerals = new string[]
    	{ "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I" };
		
		StringBuilder result = new StringBuilder();
			
			
			for (int i = 0; i < 13; i++)
		{
    		while (number >= values[i]){
        		number -= values[i];
        		result.Append(numerals[i]);
    		}
		}
		
		return result.ToString();
	}
			
	public static void Main()
	{
		Console.WriteLine(NumberToRoman(13));
	}	
}
